src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IUsingStatement.cs

```diff
@@ -916,5 +916,1461 @@ public static void M1()
+              Conversion: CommonConversion (Exists: True)
+              Operand: 
+                IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: dynamic, IsImplicit) (Syntax: 'GetDisposable() ?? input')
+
+    Next (Regular) Block[B5]
+        Entering: {R1} {R2}
+
+.try {R1, R2}
+{
+    Block[B5] - Block
+        Predecessors: [B4]
+        Statements (1)
+            IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'b = true;')
+              Expression: 
+                ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Boolean) (Syntax: 'b = true')
+                  Left: 
+                    IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.Boolean) (Syntax: 'b')
+                  Right: 
+                    ILiteralOperation (OperationKind.Literal, Type: System.Boolean, Constant: True) (Syntax: 'true')
+
+        Next (Regular) Block[B9]
+            Finalizing: {R3}
+            Leaving: {R2} {R1}
+}
+.finally {R3}
+{
+    Block[B6] - Block
+        Predecessors (0)
+        Statements (0)
+        Jump if True (Regular) to Block[B8]
+            IIsNullOperation (OperationKind.IsNull, Type: System.Boolean, IsImplicit) (Syntax: 'GetDisposable() ?? input')
+              Operand: 
+                IFlowCaptureReferenceOperation: 2 (OperationKind.FlowCaptureReference, Type: System.IDisposable, IsImplicit) (Syntax: 'GetDisposable() ?? input')
+
+        Next (Regular) Block[B7]
+    Block[B7] - Block
+        Predecessors: [B6]
+        Statements (1)
+            IInvocationOperation (virtual void System.IDisposable.Dispose()) (OperationKind.Invocation, Type: System.Void, IsImplicit) (Syntax: 'GetDisposable() ?? input')
+              Instance Receiver: 
+                IFlowCaptureReferenceOperation: 2 (OperationKind.FlowCaptureReference, Type: System.IDisposable, IsImplicit) (Syntax: 'GetDisposable() ?? input')
+              Arguments(0)
+
+        Next (Regular) Block[B8]
+    Block[B8] - Block
+        Predecessors: [B6] [B7]
+        Statements (0)
+        Next (StructuredExceptionHandling) Block[null]
+}
+
+Block[B9] - Exit
+    Predecessors: [B5]
+    Statements (0)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void UsingFlow_07()
+        {
+            string source = @"
+class P
+{
+    void M(MyDisposable input, bool b)
+/*<bind>*/{
+        using (GetDisposable() ?? input)
+        {
+            b = true;
+        }
+    }/*</bind>*/
+
+    MyDisposable GetDisposable() => throw null;
+}
+
+class MyDisposable
```

Comment:
Nit: rename to indicate that it, in fact, is _not_ disposable and wasn't intended to be.